client/src/app/users/add-user.component.spec.ts

```diff
@@ -159,8 +166,18 @@ describe('AddUserComponent', () => {
       expect(ageControl.hasError('max')).toBeTruthy();
     });
 
-    it('should not allow an age to contain non-digits', () => {
-      ageControl.setValue('123x567');
+    it('should not allow an age to be non-numeric', () => {
+      // The HTML input field itself checks to make sure that the value is
+      // numeric. Angular's form controls aren't even involved with this step.
```

Comment:
Maybe this is an argument for _not_ separating out the form controls?

Reply:
I don't think I quite understand what you're saying. Could you speak more to that point?

Reply:
Ignore this. Things have apparently changed a fair bit since I left his comment and its not relevant anymore. 😀